perfkitbenchmarker/providers/aws/aws_network.py

```diff
@@ -62,8 +62,32 @@ class AwsFirewall(network.BaseFirewall):
 
   def __init__(self):
     self.firewall_set = set()
+    self.firewall_icmp_set = set()
     self._lock = threading.Lock()
 
+  def AllowIcmp(self, vm):
+    """Opens the ICMP protocol on the firewall.
+    Args:
+      vm: The BaseVirtualMachine object to open the ICMP protocol for.
+    """
+    source = '0.0.0.0/0'
+    # type, code, region, group_id, cidr
```

Comment:
This doesn't match the fields of the tuple?

Reply:
In the aws cli, when specifying an ICMP firewall rule, the from-port and to-port numbers actually specify the type and code of ICMP packets allowed, with -1 meaning all types/codes.

https://docs.aws.amazon.com/cli/latest/reference/ec2/authorize-security-group-ingress.html
https://www.iana.org/assignments/icmp-parameters/icmp-parameters.xhtml

Reply:
Ahhh. Both the API is confusing (can you include a note that port=type and link to the CLI) and you've got `code` in the tuple, but not in `authorize_cmd`.  Since we've got separate sets now, you should just be able to drop it from the tuple, right?

Reply:
I've inserted a comment explaining the port=type connection and a included a link. 

I was going to leave the type and code in the tuple in case anyone wanted to add the ability to specify the type and code in the future, but I've removed it now for the sake of simplicity.